src/cpp/b2w.cpp

```diff
@@ -130,8 +141,8 @@ int main(int argc, char* argv[])
                 exit(EXIT_FAILURE);
         }
     }
-    if (argc < 11 || argc > 12) {
-        std::fprintf(stderr, "%s", help_string);
+    if ((argc - optind) < 2 or (argc - optind) > 3) {
```

Comment:
Is this condition correct?  `optind` is "the index of the next element to be processed in argv".
So if in the end argc == optind it should be ok (but it would raise an error with the current code).

Reply:
`optind` is indeed the _next_ element to be processed in argv.

So after processing all the parameters, it should be `argc == optind` _if_ there _is no more_ parameters left.

As the syntax for `b2w` is:

    char help_string[] =
        "\nUsage: b2w [options] <in.bam> <in.fasta> region\n\nOptions:\n\t-w: window length "
        "(INT)\n\t-i: increment (INT)\n\t-m: minimum overlap (INT)\n\t-x: max reads starting at a "
        "position (INT)\n\t-c: coverage threshold. Omit windows with low coverage (INT)\n\t"
        "-d: drop SNVs that are adjacent to insertions/deletions (alternate behaviour)\n\t-h: show this help\n\n";

We need at least two (bam file and fasta file) or three (optional region) parameters.
That's what initially was tested with 11 or 12, I am trying to make it a little bit more obvious, back when I de-duplicated the code and ported to HTSlib, to fix some mis-match in behaviour between `b2w` and `fil`.

That's some very old-school classic C code (look who is on the dev list :-P ) that got slightly C++ified in more recent time.